### PR TITLE
LIA-64531-render custom themes only after lia version 22.7

### PR DESCRIPTION
**Pull request by gayathri-s-khoros**
Support rendering custom themes via `li serve-sass` cmd only after lia version 22.7

### CODE REVIEW

File: lib/skin.js [hunk 1]
```diff
@@ -290,31 +290,35 @@ module.exports = function (gulp, gutil) {
...
-                        resolve(isThemeEnabled);
-                    });
-              })
-            }
-            
-            let syncFunc = sp(asyncFunction)
-            const value = syncFunc(parentId)
-            if(value === 'true') {
+          if (parentId.includes("theme")) {
+            let asyncFunction = (parentId1) => {
```
Comment (dhalbrook) [inline lib/skin.js]: why is asyncFunction a _let_ instead of a _const_?  Is it reassigned somwhere?

File: lib/skins.js [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 'use strict';
 var SKIN_FILES_PATTERN = '**/*(*.css|*.scss|*.sass|*.ftl|*.png|*.svg|*.jpg|*.jpeg|skin.properties)';
+const sp = require('synchronized-promise')
```
Comment (dhalbrook) [inline lib/skins.js:9]: Is **synchronized-promise** already in package.json?

File: lib/skins.js [hunk 3]
```diff
@@ -292,13 +293,35 @@ module.exports = function (gulp, gutil) {
     });
   }
 
+  function checkVersionForThemeSupport() {
+    var server = require('../lib/server.js')(gulp, gutil);
+    var versionCheck = require('../lib/version-check.js')(gulp, gutil);
+    let asyncFn = () => {
```
Comment (dhalbrook) [inline lib/skins.js]: This _let_ and the following ones can be _const_ I think

File: lib/version-check.js [hunk 4]
```diff
@@ -77,9 +90,38 @@ module.exports = function (gulp, gutil) {
...
+    if (server.serverUrl() === undefined) {
+      var errMessage = 'A server URL is required in your configuration. ';
+      callbackOrThrowError(errorCallback, errMessage);
+    }
+    validate(server.serverUrl(), server.pluginToken(), opts, function() {
+      var localSkins = fs.readdirSync(skinsBaseDir).forEach(function(file) {
+        var skinDir = path.resolve(skinsBaseDir, file);
+        var skinPropFilePath = path.resolve(skinDir, "skin.properties");
+        if (isFile(skinPropFilePath)) {
+          var fileVal = fs.readFileSync(skinPropFilePath).toString();
```
Comment (dhalbrook) [inline lib/version-check.js]: You probably want some sort of try/catch around any sort of file read operation.

File: lib/skins.js (hunk 2 above)
Comment (gayathri-s-khoros) [reply lib/skins.js:9]: yes, i added this lib to package.json as part of this PR - https://github.com/lithiumtech/lithium-sdk/pull/141

File: lib/skin.js (hunk 1 above)
Comment (gayathri-s-khoros) [reply lib/skin.js]: I will change it to const

File: lib/skins.js (hunk 3 above)
Comment (gayathri-s-khoros) [reply lib/skins.js]: okay, let me change it to const

File: lib/version-check.js [hunk 5]
```diff
@@ -34,18 +48,17 @@ module.exports = function (gulp, gutil) {
...
           process.exitCode = 1;
-          callbackOrThrowError(errorCallback, 'Invalid version check response ' + version);
-          return;
+          return callbackOrThrowError(errorCallback, 'Invalid version check response ' + version);
         }
         var versionOnServer = parseInt(matches[1]);
         var minorVersionOnServer = parseInt(matches[2]);
         if (versionOnServer < serverVersion.supportedVersionMajor ||
-          (versionOnServer === serverVersion.supportedVersionMajor &&
+          (versionOnServer == serverVersion.supportedVersionMajor &&
```
Comment (MuttuKS) [inline lib/version-check.js]: `===` check is safer than `==`

Comment (gayathri-s-khoros) [reply lib/version-check.js]: yes, but `===` is comparing the data type as well, which ended up giving false here even when the above 2 values were equal, so had to change it to `==`

Comment (pujapriya17) [reply lib/version-check.js]: You could try to convert one of them so that both of them have same type and then use '==='

Comment (gayathri-s-khoros) [reply lib/version-check.js]: okay, will change accordingly to use `===`

**Review by MuttuKS — APPROVED** [review]

**Review by NarendraPrabhu — APPROVED** [review]

**Pull request merged**